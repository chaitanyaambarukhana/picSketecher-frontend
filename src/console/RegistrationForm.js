import React, { useState } from "react";
import "../assets/stylesheets/LoginForm.css";
import { titleCase } from "title-case";
const RegistrationForm = ({ handleTabLoginChange }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [message, setMessage] = useState("");
  const [error, setError] = useState("");

  const handleRegistration = (event) => {
    event.preventDefault();
    console.log("Registration Button clicked");
    console.log("keys", email, password, firstName, lastName);
    const request = {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        email: email,
        password: password,
        firstname: firstName,
        lastname: lastName,
      }),
    };
    fetch("http://localhost:8000/register/", request)
      .then((res) => res.json())
      .then((data) => {
        if (data.success == true) {
          setError("");
          setMessage(data.message);
          setEmail("");
          setFirstName("");
          setLastName("");
          setPassword("");
        } else {
          setMessage("");
          setError(data.message);
        }
      });
  };

  const handleLogin = () => {
    handleTabLoginChange(false);
  };

  return (
    <div className="form-fields">
      {message && <div className="success">{titleCase(message)}</div>}
      {error && <div className="error">{titleCase(error)}</div>}

      <div className="form-field">
        <label htmlFor="firstName" id="firstName" className="label">
          First Name
        </label>{" "}
        <input
          type="text"
          className="text-input"
          id="firstName"
          value={firstName}
          onChange={(event) => {
            setFirstName(event.target.value);
          }}
        />
      </div>
      <div className="form-field">
        <label htmlFor="lastName" id="lastName" className="label">
          Last Name
        </label>{" "}
        <input
          type="text"
          className="text-input"
          id="lastName"
          value={lastName}
          onChange={(event) => {
            setLastName(event.target.value);
          }}
        />
      </div>
      <div className="form-field">
        <label htmlFor="email" id="email" className="label">
          Email
        </label>{" "}
        <input
          type="email"
          className="text-input"
          id="email"
          value={email}
          onChange={(event) => {
            setEmail(event.target.value);
          }}
        />
      </div>

      <div className="form-field">
        <label htmlFor="password" id="password" className="label">
          Password
        </label>{" "}
        <input
          type="password"
          className="text-input"
          id="password"
          value={password}
          onChange={(event) => {
            setPassword(event.target.value);
          }}
        />
      </div>
      <button className="button-1" onClick={handleRegistration}>
        SIGN UP
      </button>
      <p className="footer">
        Already have account?{" "}
        <button onClick={handleLogin} className="sign-up-now">
          Log In
        </button>
      </p>
    </div>
  );
};
export default RegistrationForm;
