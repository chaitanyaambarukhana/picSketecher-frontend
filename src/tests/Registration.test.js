import "jsdom-global/register";
import { shallow, mount } from "enzyme";
import "../../enzymeConfig.js";
import React from "react";
import renderer from "react-test-renderer";
import RegistrationForm from "../console/RegistrationForm.js";

describe("<Login/>", () => {
  const component = renderer.create(<RegistrationForm />).toJSON();
  it("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render a email input tag", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(wrapper.find('input[name="email"]').exists()).toBe(true);
  });

  it("should render a password input tag", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(wrapper.find('input[name="password"]').exists()).toBe(true);
  });

  it("should render a submit button", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(wrapper.find('button[name="button"]').exists()).toBe(true);
  });

  it("the default value for both fields should be empty", () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(wrapper.find('input[name="email"]').prop("value")).toBe("");
    expect(wrapper.find('input[name="password"]').prop("value")).toBe("");
  });

  it("on change of value in the field, the state of that field in the component should be updated", () => {
    const wrapper = shallow(<RegistrationForm />);

    wrapper.find('input[name="firstName"]').simulate("change", {
      target: {
        value: "firstname",
      },
    });
    expect(wrapper.find('input[name="firstName"]').prop("value")).toBe(
      "firstname"
    );
    wrapper.find('input[name="lastName"]').simulate("change", {
      target: {
        value: "lastname",
      },
    });
    expect(wrapper.find('input[name="lastName"]').prop("value")).toBe(
      "lastname"
    );

    wrapper.find('input[name="email"]').simulate("change", {
      target: {
        value: "email@gmail.com",
      },
    });
    expect(wrapper.find('input[name="email"]').prop("value")).toBe(
      "email@gmail.com"
    );
    wrapper.find('input[name="password"]').simulate("change", {
      target: {
        value: "Test@1234",
      },
    });
    expect(wrapper.find('input[name="password"]').prop("value")).toBe(
      "Test@1234"
    );
  });

  //   it("on submitting, a submit handler function should be triggered on click event", () => {
  //     const fn = jest.fn();
  //     const wrapper = mount(<RegistrationForm onSubmit={fn} />);
  //     wrapper.find("form").simulate("click");
  //     expect(fn.mock.calls.length).toEqual(0);
  //   });
});
