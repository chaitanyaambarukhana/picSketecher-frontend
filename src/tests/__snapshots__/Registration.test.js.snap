// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should match the snapshot 1`] = `
<form
  className="form-fields"
  onSubmit={[Function]}
>
  
  
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="firstName"
      id="firstName"
      name="firstName"
    >
      First Name
    </label>
     
    <input
      className="text-input"
      id="firstName"
      name="firstName"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="lastName"
      id="lastName"
      name="lastName"
    >
      Last Name
    </label>
     
    <input
      className="text-input"
      id="lastName"
      name="lastName"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="email"
      id="email"
      name="email"
    >
      Email
    </label>
     
    <input
      className="text-input"
      id="email"
      name="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="form-field"
  >
    <label
      className="label"
      htmlFor="password"
      id="password"
      name="password"
    >
      Password
    </label>
     
    <input
      className="text-input"
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    className="button-1"
    name="button"
  >
    SIGN UP
  </button>
  <p
    className="footer"
  >
    Already have account?
     
    <button
      className="sign-up-now"
      onClick={[Function]}
    >
      Log In
    </button>
  </p>
</form>
`;
